src/common/ipc/type-enforced-ipc.ts

```diff
@@ -68,25 +70,53 @@ export function onceCorrect<
  * @param verifier The function to be called to verify that the args are the correct type
  */
 export function onCorrect<
-  EM extends EventEmitter,
-  L extends (event: HandlerEvent<EM>, ...args: any[]) => any
+  IPC extends EventEmitter,
+  Listener extends (event: ListenerEvent<IPC>, ...args: any[]) => any
 >({
   source,
   channel,
   listener,
   verifier,
 }: {
-  source: EM,
-  channel: string | symbol,
-  listener: L,
-  verifier: ListVerifier<Rest<Parameters<L>>>,
-}): void {
-  source.on(channel, (event, ...args: unknown[]) => {
+  source: IPC,
+  channel: string,
+  listener: Listener,
+  verifier: ListVerifier<Rest<Parameters<Listener>>>,
+}): Disposer {
+  function wrappedListener(event: ListenerEvent<IPC>, ...args: unknown[]) {
     if (verifier(args)) {
       (async () => (listener(event, ...args)))() // might return a promise, or throw, or reject
         .catch(error => logger.error("[IPC]: channel on handler threw error", { channel, error }));
     } else {
       logger.error("[IPC]: channel was emitted with invalid data", { channel, args });
     }
-  });
+  }
+
+  source.on(channel, wrappedListener);
+
+  return () => source.off(channel, wrappedListener);
+}
+
+export function handleCorrect<
+  Handler extends (event: Electron.IpcMainInvokeEvent, ...args: any[]) => any,
+>({
+  channel,
+  handler,
+  verifier,
+}: {
+  channel: string,
+  handler: Handler,
+  verifier: ListVerifier<Rest<Parameters<Handler>>>,
+}): Disposer {
```

Comment:
this part is a bit hard to follow. Could this be defined type instead?
```
{
  channel: string,
  handler: Handler,
  verifier: ListVerifier<Rest<Parameters<Handler>>> 
}
```

Reply:
Sure

Reply:
Didn't we agree to remove the verifier concept??

Reply:
I thought I saw a commit to this PR that removed it. 🤔

Reply:
I did, this function is an internal one

Reply:
Ah, OK. 👍